### PR TITLE
added support for encrypt files for 7z archives

### PR DESCRIPTION
**Pull request by silverqx**
7z archives support archive header encryption by -mhe=on command line parameter, also unit tests added

### CODE REVIEW

File: src/Archive7z.php [hunk 1]
```diff
@@ -223,6 +252,10 @@ private function decorateCmdCompress(): array
 
         if (null !== $this->password) {
             $out[] = '-p' . $this->password;
+
+            // Encrypt archive header if 7z archive
+            if ($this->encryptFilenames && (pathinfo($this->filename, PATHINFO_EXTENSION) === '7z'))
+                $out[] = '-mhe=on';
```
Comment (Gemorroj) [inline src/Archive7z.php]: plz, psr2 style.
https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

File: tests/Archive7zTest.php [hunk 2]
```diff
@@ -577,4 +587,128 @@ public function testIsValid(string $archiveName): void
...
+            self::assertRegExp('/Can not open encrypted archive\. Wrong password\?/', $e->getMessage());
+            self::assertEquals($e->getProcess()->getExitCode(), 2);
+        }
+    }
+
+
+    public function testAddEntryPasswdEncFiles(): void
+    {
+        \copy($this->fixturesDir . '/test.txt', $this->tmpDir . '/file.txt');
+        $tempArchive = $this->tmpDir . '/' . 'archive7z_' . substr(md5(time()), 0, 6) . '.7z';
```
Comment (Gemorroj) [inline tests/Archive7zTest.php]: use https://www.php.net/tempnam plz

Comment (silverqx) [reply tests/Archive7zTest.php]: done

File: src/Archive7z.php (hunk 1 above)
Comment (silverqx) [reply src/Archive7z.php]: done

**Comment by Gemorroj** [comment]
thanks!

**Pull request merged**